.circleci/config.yml

```diff
@@ -352,9 +352,66 @@ commands:
                     LAST_RELEASE=${LAST_RELEASE:1}
                   fi
                   if [[ -n ${CIRCLE_BRANCH+x} ]] && [[ $CIRCLE_BRANCH != "main" ]] && [[ "$(git log -1)" = *"publish_android_snapshot"* ]]; then
-                    sed -i -e "s/^VERSION_NAME=.*/VERSION_NAME=${LAST_RELEASE}-${CIRCLE_BRANCH}-${COMMIT_SHA}-SNAPSHOT/" gradle.properties
+                    export SNAPSHOT_VERSION_NAME="${LAST_RELEASE}-${CIRCLE_BRANCH}-${COMMIT_SHA}-SNAPSHOT"
+                    # Make it available to next steps
+                    echo "export SNAPSHOT_VERSION_NAME=${SNAPSHOT_VERSION_NAME}" >> "$BASH_ENV"
+
+                    sed -i -e "s/^VERSION_NAME=.*/VERSION_NAME=${SNAPSHOT_VERSION_NAME}/" gradle.properties
                     make sdkRegistryUpload
+                  else
+                    echo "Skipping Slack notification - snapshot wasn't requested"
+                    circleci-agent step halt
```

Comment:
Nice trick I've stolen from iOS repo to stop the rest job from executing while not reporting failure